src/sentry/api/endpoints/user_authenticator_details.py

```diff
@@ -159,7 +161,18 @@ def delete(self, request: Request, user, auth_id, interface_device_id=None) -> R
                 user, ignore_backup=True
             )
             last_2fa_method = len(enrolled_methods) == 1
-            require_2fa = user.get_orgs_require_2fa().exists()
+            # TODO(hybrid-cloud): synchronize flags for org settings onto org mapping to make this RPC unnecessary?
+            require_2fa = (
+                len(
+                    organization_service.get_organizations(
+                        user_id=user.id,
+                        scope=None,
+                        flags=models.F("flags").bitor(Organization.flags.require_2fa),
```

Comment:
Should the service layer model this as a keyword argument eg. `get_organizations(user_id=user.id, require_2fa=True)`? Having to generate conditions with bitmasks before doing an RPC call feels a bit silly.

Reply:
I agree and will take a look.